src/Shared/NodeEndpointOutOfProcBase.cs

```diff
@@ -422,7 +422,11 @@ private void PacketPumpProc()
                     // it will cause the host to reject us; new hosts expect 00 and old hosts expect F5 or 06).
                     try
                     {
-                        long handshake = localReadPipe.ReadLongForHandshake(/* reject these leads */ new byte[] { 0x5F, 0x60 }, 0xFF /* this will disconnect the host; it expects leading 00 or F5 or 06 */);
+                        long handshake = localReadPipe.ReadLongForHandshake(/* reject these leads */ new byte[] { 0x5F, 0x60 }, 0xFF /* this will disconnect the host; it expects leading 00 or F5 or 06 */
+#if NETCOREAPP2_1
+                            , ClientConnectTimeout /* wait a long time for the handshake from this side */
```

Comment:
Was this a dead field before? Curious when / how it was used.

Reply:
I couldn't find any use of it before.

Reply:
(But it was basically exactly what I wanted, so I used it. There's no principled reason for it to be double the other timeout, though.)